packages/polling-controller/src/PollingController.test.ts

```diff
@@ -0,0 +1,214 @@
+    });
+    it('should error if no poll token is passed', () => {
+      jest.useFakeTimers();
+      class MyGasFeeController extends PollingController<any, any, any> {
+        executePoll = executePollMock;
+      }
+      const mockMessenger = new ControllerMessenger<any, any>();
+
+      const controller = new MyGasFeeController({
+        messenger: mockMessenger,
+        metadata: {},
+        name: 'PollingController',
+        state: { foo: 'bar' },
+      });
+      controller.start('mainnet');
+      expect(() => {
+        controller.stop(undefined as unknown as any);
+      }).toThrow('pollingToken required');
+      controller.stopAll();
+    });
+    it('should error if no poll token is found', () => {
+      jest.useFakeTimers();
+      class MyGasFeeController extends PollingController<any, any, any> {
+        executePoll = executePollMock;
+      }
+      const mockMessenger = new ControllerMessenger<any, any>();
+
+      const controller = new MyGasFeeController({
+        messenger: mockMessenger,
+        metadata: {},
+        name: 'PollingController',
+        state: { foo: 'bar' },
+      });
+      controller.start('mainnet');
+      expect(() => {
+        controller.stop('potato');
+      }).toThrow('pollingToken not found');
+      controller.stopAll();
+    });
+  });
+  describe('poll', () => {
+    it('should call executePoll if polling', async () => {
+      jest.useFakeTimers();
+
+      class MyGasFeeController extends PollingController<any, any, any> {
+        executePoll = executePollMock;
+      }
+      const mockMessenger = new ControllerMessenger<any, any>();
+
+      const controller = new MyGasFeeController({
+        messenger: mockMessenger,
+        metadata: {},
+        name: 'PollingController',
+        state: { foo: 'bar' },
+      });
+      controller.start('mainnet');
+      jest.advanceTimersByTime(1200);
+      await Promise.resolve();
+      jest.advanceTimersByTime(1200);
+      await Promise.resolve();
+      expect(executePollMock).toHaveBeenCalledTimes(2);
+    });
+    it('should continue calling executePoll when start is called again with the same networkClientId', async () => {
+      jest.useFakeTimers();
+
+      class MyGasFeeController extends PollingController<any, any, any> {
+        executePoll = executePollMock;
+      }
+      const mockMessenger = new ControllerMessenger<any, any>();
+
+      const controller = new MyGasFeeController({
+        messenger: mockMessenger,
+        metadata: {},
+        name: 'PollingController',
+        state: { foo: 'bar' },
+      });
+      controller.start('mainnet');
+      controller.start('mainnet');
+      jest.advanceTimersByTime(1200);
+      await Promise.resolve();
+      jest.advanceTimersByTime(1200);
+      await Promise.resolve();
+      expect(executePollMock).toHaveBeenCalledTimes(2);
+      controller.stopAll();
+    });
+    it('should publish polligComplete when stop is called', async () => {
```

Comment:
```suggestion
    it('should publish 'pollingComplete' when stop is called', async () => {
```

Also should this one be in the describe('stop' () => ...) block?